### DISCUSSION TITLE
Displaying Apple Smart Banner on certain pages only

### DISCUSSION THREAD

**Discussion by adamivancza** [vercel/next.js#43807] [Help]
We're trying to display the Apple Smart Banner on some pages (let's call them page A for simplicity) but not on all of them (as of now we only support Universal Linking on some pages only). So we added this code snippet to these pages:
```
{isBannerPromotionActive && config.APPLE_APPSTORE_ID && (
     <meta
       name="apple-itunes-app"
       content={`app-id=${config.APPLE_APPSTORE_ID}, app-argument=${deepLink}`}
       key="smart-banner"
     />
)}
```

However, we're seeing the following behavior in iOS on Safari:
* If the user goes straight to page A (e.g. from google goes to mysite.com/page_a) then the banner appears as expected. Although the banner remains visible all the time if the user navigates to any other page, even though that page doesn't have the banner meta tag in its source.
* If the user goes to the homepage (e.g. goes to mysite.com) and then navigates to page A then the banner is not visible.

Any idea why is this happening and how can we sort this out?

**Comment by icyJoseph** [comment]
I think the issue is that they read the initial HTML in order to do this. Assuming the meta tag is at page level and not `_app`.

If you loaded this dynamically, via JavaScript, it wouldn't show IIRC.

That's why going straight to the page they see it, however, when they navigate to the page the tag is added using JavaScript and they seem to ignore that. 

http://openradar.appspot.com/21871593

I am not sure if there's some type of trick to change this, but I do recall working with this not long ago, and in a test I just did with an iPhone, it seems to be the case as well.

> Imagine you had a `show` flag, initially false that controls the banner, and which you set to true inside use-effect client side, then the banner wouldn't show up either.

I'd say, that navigating **to, and away** from that page ought to be done via regular `a` tags, as opposed to `next/link`, in order to trigger an actual navigation. 

LINT might complain but it is a very strong edge case so you can just disable the rule for that section of the JSX.
